packages/engine/Source/Scene/CameraEventAggregator.js

```diff
@@ -219,20 +218,73 @@ function listenMouseButtonDownUp(aggregator, modifier, type) {
 
   aggregator._eventHandler.setInputAction(
     function () {
-      aggregator._buttonsDown = Math.max(aggregator._buttonsDown - 1, 0);
-      isDown[key] = false;
-      releaseTime[key] = new Date();
+      cancelMouseDownAction(getKey(type, undefined), aggregator);
+      for (const modifierName in KeyboardEventModifier) {
+        if (KeyboardEventModifier.hasOwnProperty(modifierName)) {
+          const modifier = KeyboardEventModifier[modifierName];
+          if (defined(modifier)) {
+            const cancelKey = getKey(type, modifier);
+            cancelMouseDownAction(cancelKey, aggregator);
+          }
+        }
+      }
     },
     up,
     modifier
   );
 }
 
+function cancelMouseDownAction(cancelKey, aggregator) {
+  const releaseTime = aggregator._releaseTime;
+  const isDown = aggregator._isDown;
+  if (isDown[cancelKey]) {
+    aggregator._buttonsDown = Math.max(aggregator._buttonsDown - 1, 0);
+  }
+  isDown[cancelKey] = false;
+  releaseTime[cancelKey] = new Date();
+}
+
 function cloneMouseMovement(mouseMovement, result) {
   Cartesian2.clone(mouseMovement.startPosition, result.startPosition);
   Cartesian2.clone(mouseMovement.endPosition, result.endPosition);
 }
 
+function refreshMouseDownStatus(type, modifier, aggregator) {
+  // first: Judge if the mouse is pressed
+  const isDown = aggregator._isDown;
+  let downFlag = false;
+  Object.keys(isDown).map((key) => {
+    if (
+      key.toString().startsWith(type) &&
+      isDown[key] &&
+      key.toString() !== getKey(type, modifier).toString()
+    ) {
+      downFlag = true;
+      cancelMouseDownAction(key, aggregator);
+    }
+  });
```

Comment:
1. It's not good to use `map()` when not actually _mapping_ something. `forEach` is designed for looping over each item of a list.
2. `Object.keys` always returns an array of strings, no need to `toString` the keys. To avoid having to `toString` the `getKey` value I think we should modify that function itself to always return a `string` instead of sometimes a `string` and sometimes a `number` like it does now. That would be possible by just wrapping the first assignment in a template string. Please test to make sure that change doesn't break other parts of this but I don't think it should.
3. Since you want the key and the value of the `isDown` object it may be better to loop over the `Object.entries` here as well.
4. We want the value of `getKey(type, modifier)` here and down lower in the function. We also don't need to re-calculate it for every key in `isDown`. it would be better to extract it outside the loop to a `currentKey` variable.